### PR TITLE
Cypress tests

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes
- adds a github actions workflow that will run alongside other tests
- uses cypress to run tests

Todo:
- [x] create sustainable way to structure and write the tests (Cypress reduces need for many other frameworks)
- [x] setup initial suite of tests for [important](#612 ) flows
- [ ] TBD: cache containers

## Checklist
- [x] All querysets/queries filter by Team (if applicable)
- [x] Backend tests (if applicable)

### CODE REVIEW

**Comment by mariusandra** [comment]
This is looking pretty sweet! Is there a chance you could do a quick screencast showing how this works at some point? :)

**Comment by EDsCODE** [comment]
Yep! Will add guidelines to contributing docs/somewhere related also

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout
+        uses: actions/checkout@v1
+      # Install Node.js
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+      - run: yarn install
+      
+      - name: Prepare server
+        run: docker-compose -f docker-compose.e2e.yml up -d
```
Comment (timgl) [inline .github/workflows/e2e.yml:16]: Would [something like this](https://semaphoreci.com/docs/docker/docker-layer-caching.html) work to pull the :latest image from master and then use that as a cache to build the image?

**Comment by EDsCODE** [comment]
docs [here](https://github.com/PostHog/posthog.com/pull/68)

File: cypress/support/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                .type('company')
+                .should('have.value', 'company')
+
+            cy.get('#inputName')
+                .type('name')
+                .should('have.value', 'name')
+
+            cy.get('#inputEmail')
+                .type('fake@email.com')
+                .should('have.value', 'fake@email.com')
```
Comment (timgl) [inline cypress/support/index.js:37]: If we change this to @posthog.com it'll automatically be ignored in our real_users cohort

File: cypress/support/index.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                .should('have.value', 'password')
+
+            cy.get('.btn').click()
+        }
+    })
+    cy.visit('http://localhost:8000/demo')
+    cy.visit('http://localhost:8000')
+})
+
+Cypress.on('uncaught:exception', err => {
```
Comment (EDsCODE) [inline cypress/support/index.js:60]: suggested configuration on cypress to prevent tests from completely failing from exceptions

Comment (EDsCODE) [inline cypress/support/index.js:60]: suggested configuration on cypress to prevent tests from completely failing from exceptions

**Comment by EDsCODE** [comment]
@timgl @mariusandra I still have to investigate the caching possibilities but I'd suggest you guys give it a try to see how cypress works. If the docs aren't enough guidance, I can provide a screencast. Right now the assertions are just testing for simple existence

**Comment by EDsCODE** [comment]
`bin/frontend-test-runner` will create a cypress client listening to localhost:8080 with its own fresh DB with demo data so the tests will be isolated from local dev environment

File: bin/frontend-test-runner [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -e
+
+dropdb --if-exists posthog_e2e_test
+createdb posthog_e2e_test
+DEBUG=1 DATABASE_URL=postgresql://postgres:postgres@localhost/posthog_e2e_test python manage.py migrate
```
Comment (timgl) [inline bin/frontend-test-runner:6]: Default for me locally is `postgresql://localhost:5432/posthog_e2e_test` and wasn't able to overwrite this.

Comment (EDsCODE) [reply bin/frontend-test-runner]: that's the right default. changed!

**Review by timgl — APPROVED** [review]

**Review by mariusandra — APPROVED** [review]

**Pull request merged**